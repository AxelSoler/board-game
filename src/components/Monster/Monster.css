.gridMonster {
  width: 49%;
  border: 1px solid #fff;
  height: 100px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 12px 0;
}

a {
  text-decoration: none;
  color: #fff;
}

p {
  margin: 12px 0;
}

.monstername {
  font-size: 1.5rem;
}
