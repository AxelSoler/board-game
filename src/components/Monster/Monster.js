import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Monster = (props) => {
  const { id, name } = props;

  return (
    <div id={id}>
      <NavLink
        className={({ isActive }) => (isActive ? 'link-active' : 'link')}
        to={`/Monster/${name}`}
      >
        {name}
      </NavLink>
      <p>
        MONSTER ID:
        {' '}
        {id}
      </p>
    </div>
  );
};

export default Monster;

Monster.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
};

Monster.defaultProps = {
  id: 0,
  name: 'MONSTER',
};
