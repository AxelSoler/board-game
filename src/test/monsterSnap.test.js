import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Monster from '../components/Monster/Monster';

test('monster snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Monster key="345" id={345} name="Axel" />
        ,
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
