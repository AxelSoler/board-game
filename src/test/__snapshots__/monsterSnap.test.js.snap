// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monster snapshot 1`] = `
Array [
  <div
    className="gridMonster"
    id={345}
  >
    <a
      className="link"
      href="/Monster/Axel"
      onClick={[Function]}
    >
      <p
        className="monstername"
      >
        Axel
      </p>
      <p>
        MONSTER ID:
         
        345
      </p>
    </a>
  </div>,
  ",",
]
`;
